### PR TITLE
shared lib

### PR DESCRIPTION
**Pull request by jiayuehua**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [*] enhancement

#### Description:
shared lib other than static lib in nebula

## How do you solve it?
change cmakelists.txt

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by darionyaphet** [comment]
Using the shared library to replace objects could shorten compilation time? If so what's the proportion？

File: src/daemons/CMakeLists.txt [hunk 1]
```diff
@@ -3,58 +3,58 @@
...
-    $<TARGET_OBJECTS:expression_obj>
-    $<TARGET_OBJECTS:function_manager_obj>
-    $<TARGET_OBJECTS:agg_function_manager_obj>
-    $<TARGET_OBJECTS:time_utils_obj>
-    $<TARGET_OBJECTS:datetime_parser_obj>
-    $<TARGET_OBJECTS:version_obj>
-    $<TARGET_OBJECTS:ssl_obj>
-    $<TARGET_OBJECTS:geo_index_obj>
-    $<TARGET_OBJECTS:log_monitor_obj>
+    meta_obj
```
Comment (Shylock-Hg) [inline src/daemons/CMakeLists.txt:6]: Change test executable to shared link is enough.

File: src/common/datatypes/Geography.cpp [hunk 2]
```diff
@@ -10,11 +10,11 @@
 
 #include <cstdint>
 
-#include "common/geo/GeoUtils.h"
-#include "common/geo/io/wkb/WKBReader.h"
```
[the remark this answers is not in the document]
Comment (jiayuehua) [reply src/common/datatypes/Geography.cpp:14]: common/geo and common/datatypes have circle dependency

File: src/common/datatypes/io/CMakeLists.txt [hunk 3]
```diff
@@ -34,6 +45,9 @@ nebula_add_library(
     wkb/WKBWriter.cpp
     wkb/ByteOrderDataIOStream.cpp
 )
+target_link_libraries(datatypes_obj PUBLIC base_obj s2 "${Glog_LIBRARY}" )
+add_library(wkt_wkb_io_obj ALIAS datatypes_obj)
+add_library(geo_index_obj ALIAS datatypes_obj)
```
[the remark this answers is not in the document]
Comment (jiayuehua) [reply src/common/datatypes/io/CMakeLists.txt:50]: those three also have circle dependency, So merge them to to one lib.

**Comment by jiayuehua** [comment]
> Using the shared library to replace objects could shorten compilation time? If so what's the proportion？

Its meant to reduce code size. A feature required by tech board.

**Comment by jiayuehua** [comment]
far from complete

**Pull request closed without merging**